File: aoc-2023/src/test/java/com/capital7software/aoc/aoc2023/days/Day07Test.java
package com.capital7software.aoc.aoc2023.days;

import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day07Test extends AdventOfCodeTestBase {
    private static final Logger LOGGER = LoggerFactory.getLogger(Day07Test.class);

    @BeforeEach
    void setUp() {
        var instance = new Day07();
        setupFromFile(instance.getDefaultInputFilename());
    }

    @Test
    void calculateTotalWinningsUsingJacks() {
        var instance = new Day07();
        int expected = 6440;
        var actual = instance.calculateTotalWinnings(lines, false);
        assertEquals(expected, actual, "The total winnings using Jacks " +
                "is not what was expected: " + expected);
    }

    @Test
    void calculateTotalWinningsUsingJokers() {
        var instance = new Day07();
        int expected = 5905;
        var actual = instance.calculateTotalWinnings(lines, true);
        assertEquals(expected, actual, "The total winnings using Jokers " +
                "is not what was expected: " + expected);
    }

    @Override
    protected Logger getLogger() {
        return LOGGER;
    }
}

File: aoc-2016/src/test/kotlin/com/capital7software/aoc/aoc2016aoc/days/Day07Test.kt
package com.capital7software.aoc.aoc2016aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day07Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day07Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day07()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testIpsThatSupportTls() {
    val instance = Day07()
    val expected = 2L
    val actual = instance.ipsThatSupportTls(lines)

    assertEquals(expected, actual, "The IPs that support TLS "
        + "is not what was expected: $expected")
  }

  @Test
  fun testIpsThatSupportSsl() {
    setupFromFile("inputs/input_day_07-02.txt")
    val instance = Day07()
    val expected = 3L
    val actual = instance.ipsThatSupportSsl(lines)

    assertEquals(expected, actual, "The IPs that support SSL "
        + "is not what was expected: $expected")
  }
}
